inc/az_ulib_base.h

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. 
+// See LICENSE file in the project root for full license information.
+
+/**
+ * @file az_ulib_base.h
+ *
+ * @brief   Some base definitions for the ulib.
+ */
+
+#ifndef AZ_ULIB_BASE_H
+#define AZ_ULIB_BASE_H
+
+#include "azure_macro_utils/macro_utils.h"
+#include "umock_c/umock_c_prod.h"
+
+#ifndef __cplusplus
+#include <stdbool.h>
+#include <stdint.h>
+#else
+#include <cstdint>
+extern "C" {
+#endif
+
+/**
+ * @brief   Does not wait if the resource is not available returning the proper error code.
+ */
+#define AZ_ULIB_NO_WAIT 0x00000000
+
+/**
+ * @brief   Wait forever if the resource is not available.
+ */
+#define AZ_ULIB_WAIT_FOREVER 0xFFFFFFFF
+
+#define AZ_ULIB_FLAGS_IS_SET(flag, bits) ((flag & bits) != 0)
+
+/**
+ * @brief   Version Handler
+ */
+typedef uint32_t az_ulib_version;
```

Comment:
>az_ulib_version [](start = 17, length = 15)

are we going to define a way this should be formatted? bit shifting for major, minor, update sections etc. major is maybe `x << 12`, minor `x << 8` etc.

Reply:
It was a structure, but recently they decide that the Digital Twin will use a single number to represent the version, so I followed it.

---
In reply to: [338171840](https://github.com/Azure/azure-ulib-c/pull/40#discussion_r338171840) [](ancestors = 338171840)

Reply:
gotcha sounds good to me

---
In reply to: [338178926](https://github.com/Azure/azure-ulib-c/pull/40#discussion_r338178926) [](ancestors = 338178926,338171840)